src/Microsoft.Health.Fhir.Core/Messages/Patch/ConditionalFhirPatchResourceRequest.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Messages.Upsert;
+
+namespace Microsoft.Health.Fhir.Core.Messages.Patch
+{
+    public sealed class ConditionalFhirPatchResourceRequest : ConditionalResourceRequest<UpsertResourceResponse>
```

Comment:
>ConditionalFhirPatchResourceRequest [](http://example.com/codeflow?start=24&length=35)

This and FhirPatchResourceRequest looks excessive for me.
What are they doing what ConditionalPatchResourceRequest can't do?

Reply:
@mattpasco-microsoft can you comment on this one?

Reply:
Agreed, pushed some changes to reduce complexity so the only duplication is in the patch service itself